transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -153,7 +150,10 @@ public ChannelHandlerContext fireUserEventTriggered(Object event) {
     @Override
     public ChannelHandlerContext fireChannelRead(Object msg) {
         AbstractChannelHandlerContext next = findContextInbound();
-        ReferenceCountUtil.touch(msg, next);
+        if (ResourceLeakDetector.ENABLED) {
```

Comment:
@ninja- so do you see any perf improvements ? Can we provide it with a micro benchmark (via jmh) ?